packages/core/src/apiml/Services.ts

```diff
@@ -0,0 +1,244 @@
+                                conflictTypes: [],
+                                basePathsByPlugin: {}
+                            };
+                            profInfos.push(profInfo);
+                        }
+
+                        if (!profInfo.basePaths.includes(apiInfo.basePath)) {
+                            if (apiInfo.gatewayUrl === config.gatewayUrl || apiInfo.defaultApi) {
+                                const numGatewayUrls = new Set(new Array(...profInfo.pluginConfigs)
+                                    .filter(({ gatewayUrl }) => gatewayUrl)
+                                    .map(({ gatewayUrl }) => gatewayUrl)).size;
+                                profInfo.basePaths.splice(numGatewayUrls, 0, apiInfo.basePath);
+                            } else {
+                                profInfo.basePaths.push(apiInfo.basePath);
+                            }
+                        }
+
+                        profInfo.pluginConfigs.add(config);
+                        profInfo.basePathsByPlugin[config.pluginName] = new Set([
+                            ...(profInfo.basePathsByPlugin[config.pluginName] || []),
+                            apiInfo.basePath
+                        ]);
+                    }
+                }
+            }
+        }
+
+        // Find conflicts in profile info array
+        for (const profInfo of profInfos) {
+            // If multiple CLI plug-ins require different base paths for the
+            // same service ID and CLI profile type, we have a conflict because
+            // the plugins expect different base paths and may be incompatible.
+
+            // Find the intersection of base path sets for all CLI plug-ins
+            const basePathSets = Object.values(profInfo.basePathsByPlugin).map(set => new Array(...set));
+            const commonBasePaths = basePathSets.reduce((a, b) => a.filter(x => b.includes(x)));
+            if (commonBasePaths.length === 0) {
+                profInfo.conflictTypes.push("basePaths");
+            } else {
+                profInfo.basePaths = profInfo.basePaths.filter(basePath => commonBasePaths.includes(basePath));
+            }
+            delete profInfo.basePathsByPlugin;
+
+            // If multiple profile infos have the same type, we have a conflict
+            // because we don't know which profile should be the default.
+            if (profInfos.filter(({ profType }) => profType === profInfo.profType).length > 1) {
+                profInfo.conflictTypes.push("profType");
+            }
+        }
+
+        return profInfos;
+    }
+
+    /**
+     * Converts apiml profile information to team config profile objects to be stored in a zowe.config.json file
+     * @param profileInfoList List of apiml profiles
+     * @returns List of config profile objects
+     * @memberof Services
+     */
+    public static convertApimlProfileInfoToProfileConfig(profileInfoList: IApimlProfileInfo[]): IConfig {
```

Comment:
This function does not throw any errors, but should we at least default the `profileInfoList` to an empty array `[]`?
Or rather do `profileInfoList?.forEach(...)` in case someone provides `null|undefined`

Reply:
Maybe a unit tests should be added for such scenario?

Reply:
Went with `profileInfoList?.forEach(...)`
Added a test for this